experimental/segmenter/src/word.rs

```diff
@@ -43,17 +26,30 @@ pub type WordBreakIteratorUtf16<'l, 's> = RuleBreakIterator<'l, 's, WordBreakTyp
 /// encodings. Please see the [module-level documentation](crate) for its usages.
 pub struct WordBreakSegmenter {
     payload: DataPayload<WordBreakDataV1Marker>,
+    dictionary_payload: DataPayload<UCharDictionaryBreakDataV1Marker>,
 }
 
 impl WordBreakSegmenter {
     pub fn try_new<D>(provider: &D) -> Result<Self, DataError>
     where
-        D: ResourceProvider<WordBreakDataV1Marker> + ?Sized,
+        D: ResourceProvider<WordBreakDataV1Marker>
+            + ResourceProvider<UCharDictionaryBreakDataV1Marker>
+            + ?Sized,
     {
         let payload = provider
             .load_resource(&DataRequest::default())?
             .take_payload()?;
-        Ok(Self { payload })
+
+        // TODO: Use `provider` parameter after we support loading dictionary data from the
+        // production-ready providers.
+        let inv_provider = icu_provider::inv::InvariantDataProvider;
+        let dictionary_payload = inv_provider
```

Comment:
Question: It seems that line-break always uses LSTM, and word-break always uses dictionary. Are you planning to allow both line-break and word-break to use either LSTM or dictionary?

Reply:
Yes, I will use both at ultimately.